src/System.Data.SqlClient/src/Interop/SNINativeMethodWrapper.cs

```diff
@@ -10,6 +10,7 @@ internal static class SNINativeMethodWrapper
     {
         private const string SNI = "sni.dll";
 
+#if !MANAGED_SNI
```

Comment:
In general for corefx we've been trying to move away from having these kinds of ifdefs sprinkled throughout the code.  Instead, we've opted for a scheme where we have different files containing the different implementations (either as static members called to or members on partial classes), and then the different builds include the files they need.  We should consider such an approach here as well.